wordless/helpers/component_helper.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+namespace Wordless;
+
+use \Symfony\Component\Validator\Validation;
+use \Symfony\Component\Validator\Mapping\ClassMetadata;
+use \Symfony\Component\Validator\Constraints as Assert;
+
+class ComponentValidationException extends \Exception {}
+
+// Implementation example
+
+// use Symfony\Component\Validator\Constraints as Assert;
+// use Symfony\Component\Validator\Context\ExecutionContextInterface;
+
+// class MyComponent extends \Wordless\Component {
+//     public $post;
+
+//     public static function loadValidatorMetadata($metadata)
+//     {
+//         $metadata->addPropertyConstraint('post', new Assert\Type(WP_Post::class));
+//         $metadata->addConstraint(new Assert\Callback('validate_post_type'));
+//     }
+
+//     public function validate_post_type(ExecutionContextInterface $context, $payload)
+//     {
+//         if( !in_array($this->post->post_type, ['post']) ) {
+//             $context->buildViolation('WP_Post passed to this component must be of type "post"')
+//                 ->atPath('post')
+//                 ->addViolation();
+//         }
+//     }
+// }
+//
+// Then in PUG
+//
+// mixin MyComponent($component)
+//   h1= $component->post->post_title
+//
+// - $component = new MyComponent(['post' => get_post(1)])
+// +MyComponent($component)
+//
+// More docs about Symphony\Component\Validator @ https://symfony.com/doc/current/validation.html#constraints
+abstract class Component {
+
+    protected $locals = [];
+    private $validator = null;
+
+    function __construct(array $locals) {
+        $this->validator = Validation::createValidatorBuilder()->addMethodMapping('loadValidatorMetadata')->getValidator();
+        $this->locals = $locals;
+        try {
+            $this->setProperties();
+            $this->validate();
+        } catch (ComponentValidationException $e) {
+            if ( 'production' === ENVIRONMENT ) {
```

Comment:
In the PR's description you say "Validation errors in production
By default no exception will be raised in production/staging"

maybe i'm wrong, but in this way staging acts as if it's development

Reply:
Yep, you're right: we have `staging` as possible value elsewhere in the code and in the docs. Sweet catch!

Reply:
BTW the problem is in teh description: I actually **want** to drop exception when on staging. I think it is so important. So the code is ok and my description inaccurate.

I'm going to resolve this one! Thanks again.